src/sentry/preprod/api/endpoints/organization_preprod_artifact_assemble.py

```diff
@@ -113,6 +122,11 @@ def post(self, request: Request, project: Project) -> Response:
             if error_message:
                 return Response({"error": error_message}, status=400)
 
+            # Support a limited subset of providers
+            provider = data.get("provider")
+            if provider is not None and provider not in SUPPORTED_VCS_PROVIDERS:
+                return Response({"error": "Unsupported provider"}, status=400)
```

Comment:
i think unsupported vcs provider would be more clear.
```suggestion
                return Response({"error": "Unsupported vcs provider"}, status=400)
```